src/python/grpcio/grpc/_common.py

```diff
@@ -97,3 +98,27 @@ def serialize(message, serializer):
 def deserialize(serialized_message, deserializer):
   return _transform(serialized_message, deserializer,
                     'Exception deserializing message!')
+
+
+class CleanupThread(threading.Thread):
+  """A wrapper around thread for custom join behavior.
+  
+  On Interpreter exit, Python will attempt to join outstanding threads
+  prior to garbage collection.  We may need to do additional cleanup, and
+  we acomplish this by wrapping the join() method.
+  """
+  
+  def __init__(self, group=None, target=None, join_target=None, name=None,
```

Comment:
The behavior to be called at join-time should be a required (and non-`None`) construction parameter because if it's ever `None` this class has no point in being used at that particular use-site.

Also I recommend just taking a single no-arg parameter (call it "behavior"?) and leave it to use-sites to wrap their args and kwargs in a `lambda` or custom function.